js/main.js

```diff
@@ -1259,6 +1258,14 @@ function makeHeroes() {
 function assignMethods() {
     for (const key in gameData.taskData) {
         let task = gameData.taskData[key]
+
+        if (!('xpLog' in task) && ('xpBigInt' in task)){
+            if (typeof task.xpBigInt === "string" && task.xpBigInt.includes("e"))
+                task.xpBigInt = exponentialToRawNumberString(task.xpBigInt)
+
+            task.xpLog = task.xpBigInt.length + Math.log10("0." + task.xpBigInt.substring(0, 15)) // This line from StackOverflow

```

Comment:
Which StackOverflow post did you take this line from?

Reply:
I pulled that line from the top answer of https://stackoverflow.com/questions/70382306/logarithm-of-a-bigint